test/integration/puppetlabs/services/jruby/class_info_test.clj

```diff
@@ -0,0 +1,229 @@
+                                                env-1-mod-2-dir-and-manifests])}]
+              (testing "all environments"
+                (is (= expected-envs-info (get-class-info))))
+              (testing "one environment by name"
+                (is (= (expected-envs-info "env1")
+                       (get-class-info-for-env "env1"))
+                    "Unexpected info retrieved for 'env1'")
+                (is (= (expected-envs-info "env2")
+                       (get-class-info-for-env "env2"))
+                    "Unexpected info retrieved for 'env2'"))))
+
+          (testing "changes to manifest content for"
+            (fs/delete-dir env-1-mod-2-dir)
+            (let [foo-manifest (.getAbsolutePath (fs/file env-1-dir
+                                                          "manifests"
+                                                          "foo.pp"))
+                  _ (create-file foo-manifest "class foo () {} \n")
+                  expected-envs-info {"env1" {foo-manifest
+                                              [{"name" "foo"
+                                                "params" []}]}
+                                      "env2" (expected-manifests-info
+                                               [env-2-dir-and-manifests
+                                                env-1-mod-1-dir-and-manifests])}]
+              (testing "all environments"
+                (is (= expected-envs-info (get-class-info))))
+              (testing "one environment by name"
+                (is (= (expected-envs-info "env1")
+                       (get-class-info-for-env "env1"))
+                    "Unexpected info retrieved for 'env1'")
+                (is (= (expected-envs-info "env2")
+                       (get-class-info-for-env "env2"))
+                    "Unexpected info retrieved for 'env2'"))))
+
+          (testing "changes to environments for"
+            (fs/delete-dir env-1-dir)
+            (let [_ (create-env env-3-dir-and-manifests)
+                  expected-envs-info {"env2" (expected-manifests-info
+                                               [env-2-dir-and-manifests])
+                                      "env3" (expected-manifests-info
+                                               [env-3-dir-and-manifests
+                                                base-mod-1-and-manifests])}]
+              (testing "all environments"
+                (is (= expected-envs-info (get-class-info))))
+              (testing "one environment by name"
+                (is (= (expected-envs-info "env2")
+                       (get-class-info-for-env "env2"))
+                    "Unexpected info retrieved for 'env2'")
+                (is (= (expected-envs-info "env3")
+                       (get-class-info-for-env "env3"))
+                    "Unexpected info retrieved for 'env3'"))))
+
+          (testing "all environments by key, without serialization"
```

Comment:
I mentioned above that the `testing` sub-blocks in this `deftest` are intentionally ordered, not meant to be run in isolation.  The final two blocks here, though, I think could be separated out into their own `deftest`, though - assuming that some of the setup from this `deftest` were repeated.  I didn't do that initially because that would involve initializing even more JRubyPuppet instances for the additional tests.  I was thinking about the performance overhead that comes with doing that and the slow performance that we're already seeing with this suite overall.